### PR TITLE
Add tokens for diff blob

### PR DESCRIPTION
**Pull request by lukasoppermann**
### Added
- `--diffBlob-hunkNum-fgColor-rest`
- `--diffBlob-hunkNum-fgColor-hover`
- `--diffBlob-hunkNum-bgColor-rest`
- `--diffBlob-hunkNum-bgColor-hover`
- `--diffBlob-additionWord-fgColor`
- `--diffBlob-deletionWord-fgColor`
- `--diffBlob-emptyNum-bgColor`
- `--diffBlob-emptyLine-bgColor`

### Renamed:
- `--diffBlob-addition-fgColor-text` -> `--diffBlob-additionLine-fgColor`
- `--diffBlob-addition-fgColor-num` -> `--diffBlob-additionNum-fgColor`
- `--diffBlob-addition-bgColor-num` -> `--diffBlob-additionNum-bgColor`
- `--diffBlob-addition-bgColor-line` -> `--diffBlob-additionLine-bgColor`
- `--diffBlob-addition-bgColor-word` -> `--diffBlob-additionWord-bgColor`
- `--diffBlob-deletion-fgColor-text` -> `--diffBlob-deletionLine-fgColor`
- `--diffBlob-deletion-fgColor-num` -> `--diffBlob-deletionNum-fgColor`
- `--diffBlob-deletion-bgColor-num` -> `--diffBlob-deletionNum-bgColor`
- `--diffBlob-deletion-bgColor-line` -> `--diffBlob-deletionLine-bgColor`
- `--diffBlob-deletion-bgColor-word` -> `--diffBlob-deletionWord-bgColor`

### Removed

- `--diffBlob-hunk-bgColor-num`

### CODE REVIEW

**Comment by lukasoppermann** [comment]
@langermank 

Technically we would need to move the `num` part etc. more to the front:

```css
  --diffBlob-hunkText-fgColor
  --diffBlob-hunkNum-fgColor-rest
  --diffBlob-hunkLine-bgColor
```

Any thoughts about this?

Also I would like to change `text` to `line` to be aligned to the other:
```css
  --diffBlob-hunkLine-fgColor
  --diffBlob-hunkLine-bgColor
  ```

File: src/tokens/functional/color/dark/app-dark.json5 [hunk 1]
```diff
@@ -102,30 +114,29 @@
...
-      fgColor: {
-        text: {
+        word: {
           $value: '{fgColor.default}',
           $type: 'color',
           $extensions: {
             'org.primer.figma': {
               collection: 'mode',
-              mode: 'dark',
+              mode: 'light',
```
Comment (langermank) [inline src/tokens/functional/color/dark/app-dark.json5:139]:
```suggestion
              mode: 'dark',
```

File: src/tokens/functional/color/dark/app-dark.json5 [hunk 2]
```diff
@@ -188,19 +198,85 @@
...
+      },
+      bgColor: {
+        line: {
+          $value: '{bgColor.accent.muted}',
           $type: 'color',
           $extensions: {
             'org.primer.figma': {
               collection: 'mode',
-              mode: 'dark',
+              mode: 'light',
```
Comment (langermank) [inline src/tokens/functional/color/dark/app-dark.json5:248]:
```suggestion
              mode: 'dark',
```

File: src/tokens/functional/color/dark/app-dark.json5 [hunk 3]
```diff
@@ -188,19 +198,85 @@
...
+              },
+            },
+          },
+          hover: {
+            $value: '{fgColor.onEmphasis}',
+            $type: 'color',
+            $extensions: {
+              'org.primer.figma': {
+                collection: 'mode',
+                mode: 'light',
```
Comment (langermank) [inline src/tokens/functional/color/dark/app-dark.json5:233]:
```suggestion
                mode: 'dark',
```

File: src/tokens/functional/color/dark/app-dark.json5 [hunk 4]
```diff
@@ -188,19 +198,85 @@
...
+        },
         num: {
-          $value: '{borderColor.accent.muted}',
+          rest: {
+            $value: '{fgColor.default}',
+            $type: 'color',
+            $extensions: {
+              'org.primer.figma': {
+                collection: 'mode',
+                mode: 'light',
```
Comment (langermank) [inline src/tokens/functional/color/dark/app-dark.json5:221]:
```suggestion
                mode: 'dark',
```

File: src/tokens/functional/color/dark/app-dark.json5 [hunk 5]
```diff
@@ -188,19 +198,85 @@
...
     hunk: {
-      bgColor: {
+      fgColor: {
+        text: {
+          $value: '{fgColor.muted}',
+          $type: 'color',
+          $extensions: {
+            'org.primer.figma': {
+              collection: 'mode',
+              mode: 'light',
```
Comment (langermank) [inline src/tokens/functional/color/dark/app-dark.json5:208]:
```suggestion
              mode: 'dark',
```

File: src/tokens/functional/color/dark/app-dark.json5 [hunk 6]
```diff
@@ -188,19 +198,85 @@
...
           },
         },
+        num: {
+          rest: {
+            $value: '{base.color.blue.8}',
+            $type: 'color',
+            $extensions: {
+              'org.primer.figma': {
+                collection: 'mode',
+                mode: 'light',
```
Comment (langermank) [inline src/tokens/functional/color/dark/app-dark.json5:261]:
```suggestion
                mode: 'dark',
```

File: src/tokens/functional/color/dark/app-dark.json5 [hunk 7]
```diff
@@ -188,19 +198,85 @@
...
+              },
+            },
+          },
+          hover: {
+            $value: '{bgColor.accent.emphasis}',
+            $type: 'color',
+            $extensions: {
+              'org.primer.figma': {
+                collection: 'mode',
+                mode: 'light',
```
Comment (langermank) [inline src/tokens/functional/color/dark/app-dark.json5:273]:
```suggestion
                mode: 'dark',
```

**Review by langermank — COMMENTED** [review]
What do you think about leaving `diffBlob-hunk-bgColor-num` in for now so that this isn't a breaking change, and adding the `rest` and `hover`? Then we can remove the usages first.

**Comment by lukasoppermann** [comment]
What about this @langermank?
> 
> Technically we would need to move the `num` part etc. more to the front:
> 
> ```css
>   --diffBlob-hunkText-fgColor
>   --diffBlob-hunkNum-fgColor-rest
>   --diffBlob-hunkLine-bgColor
> ```

This means:
- `--diffBlob-addition-fgColor-num` → `--diffBlob-additionNum-fgColor`


> Any thoughts about this?
> 
> Also I would like to change `text` to `line` to be aligned to the other:
> 
> ```css
>   --diffBlob-hunkLine-fgColor
>   --diffBlob-hunkLine-bgColor
> ```

This means:
- `--diffBlob-hunk-bgColor-line` stays the same
- `--diffBlob-hunk-fgColor-text` → `--diffBlob-hunk-fgColor-line`

**Comment by lukasoppermann** [comment]
@langermank there is also an empty line, where we use `--bgColor-muted`, it is for when an empty line is removed and then it is shown in split view on one side as gray and one side red.

CleanShot 2024-10-10 at 11 40 43@2x

I am wondering how you feel about adding a token for this as well.

- `--diffBlob-emptyLine-bgColor`
- `--diffBlob-emptyNum-bgColor`

One the one side it would be more precise and easy to use. On the other side we are adding two more tokens, even though we can just use the `--bgColor-muted` token instead.

**Comment by langermank** [comment]
@lukasoppermann I like the idea of adding a token for the empty state, just make sure it references `bgColor-muted` under the hood 👍 

What's going on with VRT here? 👀

**Pull request merged**